flytepropeller/pkg/controller/nodes/array/handler.go

```diff
@@ -507,6 +508,9 @@ func (a *arrayNodeHandler) Handle(ctx context.Context, nCtx interfaces.NodeExecu
 				return handler.DoTransition(handler.TransitionTypeEphemeral, handler.PhaseInfoFailure(idlcore.ExecutionError_SYSTEM,
 					errors.BadSpecificationError, fmt.Sprintf("failed to find ArrayNode subNode task with id: '%s'", taskID), nil)), nil
 			}
+			runtimeData := taskNode.CoreTask().GetMetadata().GetRuntime()
+			// check if the literal can be offloaded based on the runtime data and the offloading config
+			canOffloadLiteral = a.literalOffloadingConfig.Enabled && a.literalOffloadingConfig.IsSupportedSDKVersion(runtimeData.GetType().String(), runtimeData.GetVersion())
```

Comment:
I thought we decided this was unnecessary, please correct me if I am misunderstanding. This portion of the code is retrieving outputs from an already executed task (or launchplan). If the task that already executed doesn't have a valid SDK version (that supports literal offloading) that is OK. The only check we need is if the actual task that is executing over an offloaded literal supports it.

Reply:
thanks  for the catch . removing the SDK version check here